src/AppInstallerRepositoryCore/Microsoft/SQLiteStorageBase.cpp

```diff
@@ -0,0 +1,35 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+#include "pch.h"
+#include "SQLiteStorageBase.h"
+#include "Schema/MetadataTable.h"
+
+namespace AppInstaller::Repository::Microsoft
+{
+    void SQLiteStorageBase::SetLastWriteTime()
+    {
+        Schema::MetadataTable::SetNamedValue(m_dbconn, Schema::s_MetadataValueName_LastWriteTime, Utility::GetCurrentUnixEpoch());
+    }
+
+    // Recording last write time based on MSDN documentation stating that time returns a POSIX epoch time and thus
+    // should be consistent across systems.
+    std::chrono::system_clock::time_point SQLiteStorageBase::GetLastWriteTime()
+    {
+        int64_t lastWriteTime = Schema::MetadataTable::GetNamedValue<int64_t>(m_dbconn, Schema::s_MetadataValueName_LastWriteTime);
+        return Utility::ConvertUnixEpochToSystemClock(lastWriteTime);
+    }
+
+    SQLiteStorageBase::SQLiteStorageBase(const std::string& target, SQLite::Connection::OpenDisposition disposition, SQLite::Connection::OpenFlags flags, Utility::ManagedFile&& indexFile) :
+        m_dbconn(SQLite::Connection::Create(target, disposition, flags)), m_indexFile(std::move(indexFile))
+    {
+        m_dbconn.EnableICU();

```

Comment:
Not clear that ICU is needed for this database.

Reply:
Removed this from base constructor and applied it only to SQLiteIndex constructor